### PR TITLE
Move core controller code from the hof-form-wizard project into here

### PR DESCRIPTION
**Pull request by lennym**
This simplifies the dependency tree as it means all controllers are in one place, so upstream project which wish to extend controllers do not need to include the entire form wizard module.

Updates the base controller to use the dynamic config options introduced in `hof-form-controller@2.0.0`.

### CODE REVIEW

File: lib/base-controller.js [hunk 1]
```diff
@@ -47,15 +50,14 @@ module.exports = class BaseController extends Controller {
...
-
     if ((req.params.action === 'edit') && completed(next)) {
       // The user is editing the form and has already completed the next
       // step, so let's check whether we should fast-forward them to the
       // confirm page
-      next = (!this.options.continueOnEdit || next === confirmStep) ?
+      next = (!req.form.options.continueOnEdit || next === confirmStep) ?
         confirmStep :
         next + '/edit';
     }
```
Comment (joefitter) [inline lib/base-controller.js:63]: This logic exists in hof-form-controller, not sure why its still here too, currently it looks like it is performing forking logic twice

Comment (lennym) [reply lib/base-controller.js:63]: I don't know either, ~I literally copied the code~ I didn't even copy it, it's just how the code was before.

Comment (joefitter) [reply lib/base-controller.js:63]: probably not in the scope of this PR though

**Review by joefitter — APPROVED** [review]

File: lib/base-controller.js (hunk 1 above)
Comment (joefitter) [reply lib/base-controller.js:63]: I'll add an issue

Comment (joefitter) [reply lib/base-controller.js:63]: oh I already did https://github.com/UKHomeOfficeForms/hof-controllers/issues/46

**Review by JoeChapman — APPROVED** [review]

**Pull request merged**